### ISSUE TITLE
[Bug]: 503 screen when Appsmith tab is open for a long time in the browser

### ISSUE THREAD

**Issue by mohanarpit** [appsmithorg/appsmith#30610]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When Appsmith app is idle for a long time, the client shows 503 error screen even though no activity took place on that tab. I suspect some underlying API is failing & the client immediately redirects to 503 & never recovers from there. We should make the API more stable and also, the handling on the client more robust.

### Steps To Reproduce

1. Open any Appsmith app
2. Walk away from the screen & let the app idle. 
3. Check back again after 3 hours. You will see a 503 error screen even though no activity took place on the tab. 

![Screenshot 2024-01-25 at 9 31 53 AM](https://github.com/appsmithorg/appsmith/assets/458946/83e9630c-b265-4e43-812c-64ae92cf0d95)

### Environment

Production

### Version

Cloud / Self-hosted

**Comment by Nikhil-Nandagopal** [comment]
@mohanarpit which team should own this?

**Comment by jbaker43** [comment]
Is there any update on this? Our team is facing the same issue.

**Comment by Nikhil-Nandagopal** [comment]
@jbaker43 this was an intermittent issue that we weren't able to reproduce. Are you able to reproduce it?

**Comment by jbaker43** [comment]
@Nikhil-Nandagopal yes. We are using appsmith for live dashboarding and every morning we are coming into the monitors displaying a 503 error.

**Comment by jbaker43** [comment]
I should add that the data is refreshing periodically about every 5-10 minutes.

**Comment by nidhi-nair** [comment]
@jbaker43 would it be possible for you to leave your network console open overnight as well, and then share the information from in there when you see the 503? Would love to understand which of our APIs is crashing. We can set up a call to discuss this further once we have some headway here.

**Comment by nidhi-nair** [comment]
@jbaker43 we need information from your network console, so before you log out for the day, please turn the console tab on and share the list from the "Network" tab on the following day. It would be great if you could also share the logs from your instance. We have steps for acquiring logs documented [here](https://docs.appsmith.com/getting-started/setup/instance-management/how-to-get-container-logs).

**Comment by jbaker43** [comment]
Will do. I will post back here @nidhi-nair Thanks for the help!

**Comment by mohanarpit** [comment]
@nidhi-nair @ApekshaBhosale I think the reason for this error screen is because of the failure of the API `/tenants/current`. In the network console, I see errors related to `FETCH_TENANT_CONFIG_ERROR`. In the code, I can see that when this API fails, we trigger a "safe crash". If we handle the error to this API differently, I think this error screen will go away.
